### PR TITLE
User Sign In/Onboarding Fixes

### PR DESCRIPTION
**Pull request by timolegros**
## Link to Issue
Closes: #8653 
Closes: #8645

## Description of Changes
- Adds a migration which updates `User.is_welcome_onboard_flow_complete`  to true if the user already has a profile name or user id set.
- Modifies `getAddressProfile` to work with empty community id arrays
- Modifies client authentication flow to ignore current or default chain when signing in.
  - This is because a chain is not required to find a specific user via address.

## Test Plan
1. Go to http://localhost:8080/vita-fast/ and create an account with a _new_ MetaMask address
2. Sign out
3. Go to http://localhost:8080/dashboard/global
4. Sign in with the same address you just created an account with
5. Sign out
6. Go to http://localhost:8080/dydx
7. Sign in with the same address
8. Join the dydx community

At no point after step 1 should it ask you to created a new account or re-complete the new onboarding flow.

To test the migration for existing users, create a new wallet address and execute the following query:
```
UPDATE "Addresses"
SET address = '[paste your wallet address]'
WHERE address = '0xFcA51ed3FdDe1c66Ed5c37f5C048cCBc9049Ed66';
```
Then attempt to sign in. You should not be prompted with an onboarding flow (the query mocks user from #8645)

## Deployment Plan

1. 

## Other Considerations

- Must be merged after #8647

### CODE REVIEW

**Review by Rotorsoft — APPROVED** [review]

File: packages/commonwealth/client/scripts/views/modals/AuthModal/useAuthentication.tsx [hunk 1]
```diff
@@ -500,7 +500,7 @@ const useAuthentication = (props: UseAuthenticationProps) => {
             })
           : selectedAddress,
       ],
-      profileChainIds: [app.activeChainId() ?? wallet.chain],
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/modals/AuthModal/useAuthentication.tsx:503]: the intention of removing is to find any matching address that already exists regardless of the chain?

File: packages/commonwealth/server/migrations/20240729220841-set-is-welcome-onboarding.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface, Sequelize) {
+    await queryInterface.sequelize.query(`
+      UPDATE "Users"
+      SET is_welcome_onboard_flow_complete = true
+      WHERE profile ->> 'name' != 'Anonymous' AND profile ->> 'name' IS NOT NULL;
```
Comment (mzparacha) [inline packages/commonwealth/server/migrations/20240729220841-set-is-welcome-onboarding.js:9]: can we use `DEFAULT_NAME` from "@hicommonwealth/shared"

**Review by mzparacha — APPROVED** [review]

**Comment by jnaviask** [comment]
@timolegros let's get these comments addressed quickly then we can merge

File: packages/commonwealth/client/scripts/views/modals/AuthModal/useAuthentication.tsx (hunk 1 above)
Comment (timolegros) [reply packages/commonwealth/client/scripts/views/modals/AuthModal/useAuthentication.tsx:503]: Yes

File: packages/commonwealth/server/migrations/20240729220841-set-is-welcome-onboarding.js (hunk 2 above)
Comment (timolegros) [reply packages/commonwealth/server/migrations/20240729220841-set-is-welcome-onboarding.js:9]: I'm not sure that I can import from libs in migrations. Additionally, migrations are meant to be correct at a specific point in time. If I use `DEFAULT_NAME` the effect of the migration may be different in the future if the variable is updated.

File: packages/commonwealth/client/scripts/views/modals/AuthModal/useAuthentication.tsx (hunk 1 above)
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/modals/AuthModal/useAuthentication.tsx:503]: Sounds good, thanks!

File: packages/commonwealth/server/migrations/20240729220841-set-is-welcome-onboarding.js (hunk 2 above)
Comment (mzparacha) [reply packages/commonwealth/server/migrations/20240729220841-set-is-welcome-onboarding.js:9]: makes sense, thanks.

**Pull request merged**